apps/server/src/modules/tool/external-tool/controller/api-test/tool.api.spec.ts

```diff
@@ -948,4 +955,186 @@ describe('ToolController (API)', () => {
+
+				const response: Response = await loggedInClient.get(
+					`medium/${userLicence.mediumId}/media-source/BILDUNGSLOGIN/${mediaSourceEntity.sourceId}/metadata`
+				);
+
+				expect(response.statusCode).toEqual(HttpStatus.OK);
+				expect(response.body).toEqual({
+					name: biloMediaMetaData.data.title,
+					description: biloMediaMetaData.data.description,
+					publisher: biloMediaMetaData.data.publisher,
+					logoUrl: biloMediaMetaData.data.cover.href,
+					previewLogoUrl: biloMediaMetaData.data.coverSmall.href,
+					modifiedAt: new Date(biloMediaMetaData.data.modified).toISOString(),
+				});
+			});
+		});
+
+		describe('when mediumId not valid', () => {
+			const setup = async () => {
+				const mediaSourceEntity = mediaSourceEntityFactory.withBiloFormat().build();
+				const userLicence = mediaUserLicenseEntityFactory.build({ mediaSource: mediaSourceEntity });
+				const { superheroUser, superheroAccount } = UserAndAccountTestFactory.buildSuperhero({}, [
+					Permission.MEDIA_SOURCE_ADMIN,
+				]);
+				await em.persistAndFlush([superheroAccount, superheroUser, userLicence, mediaSourceEntity]);
+				em.clear();
+
+				const loggedInClient: TestApiClient = await testApiClient.login(superheroAccount);
+
+				return { loggedInClient, mediaSourceEntity };
+			};
+
+			it('should not return the metadata of media source', async () => {
+				const { loggedInClient, mediaSourceEntity } = await setup();
+
+				const response: Response = await loggedInClient.get(
+					`medium/some-id/media-source/BILDUNGSLOGIN/${mediaSourceEntity.sourceId}/metadata`
+				);
+
+				expect(response.statusCode).toEqual(HttpStatus.NOT_FOUND);
+			});
+		});
+
+		describe('when media source not valid', () => {
+			const setup = async () => {
+				const authEndpoint = `https://oauth-token-url.com/`;
+				const baseUrl = `https://oauth-base-url.com/`;
+				const mediaSourceEntity = mediaSourceEntityFactory.withBiloFormat({ authEndpoint, baseUrl }).build();
+				const userLicence = mediaUserLicenseEntityFactory.build({ mediaSource: mediaSourceEntity });
+
+				const { superheroUser, superheroAccount } = UserAndAccountTestFactory.buildSuperhero({}, [
+					Permission.MEDIA_SOURCE_ADMIN,
+				]);
+				await em.persistAndFlush([superheroAccount, superheroUser, userLicence, mediaSourceEntity]);
+				em.clear();
+
+				axiosMock.onPost(new RegExp(authEndpoint)).reply<OauthTokenResponse>(HttpStatus.OK, {
+					id_token: 'idToken',
+					refresh_token: 'refreshToken',
+					access_token: 'accessToken',
+				});
+				const responses: BiloMediaQueryResponse[] = biloMediaQueryResponseFactory.buildList(2);
+
+				axiosMock.onPost(new RegExp(baseUrl + '.*')).replyOnce(HttpStatus.OK, responses);
+
+				const loggedInClient: TestApiClient = await testApiClient.login(superheroAccount);
+
+				return { loggedInClient, userLicence, mediaSourceEntity, biloMediaMetaData: responses[0] };
+			};
+
+			it('should not return the metadata of undefined media source', async () => {
```

Comment:
it should not return the metadata of undefined media source, but what should it return 🤔.
maybe something like `should return not found` would be better.